data/googlecloudstorage/v0/read_objects.go

```diff
@@ -47,3 +57,80 @@ type Attributes struct {
 	StorageClass       string
 }
 
+func readObjects(input ReadInput, client *storage.Client) (ReadOutput, error) {
+	bucketName := input.BucketName
+	ctx := context.Background()
+	query := &storage.Query{
+		Delimiter:                input.Delimiter,
+		Prefix:                   input.Prefix,
+		Versions:                 input.Versions,
+		StartOffset:              input.StartOffset,
+		EndOffset:                input.EndOffset,
+		IncludeTrailingDelimiter: input.IncludeTrailingDelimiter,
+		MatchGlob:                input.MatchGlob,
+		IncludeFoldersAsPrefixes: input.IncludeFoldersAsPrefixes,
+	}
+
+	it := client.Bucket(bucketName).Objects(ctx, query)
+
+	output := ReadOutput{
+		TextObjects:     []TextObject{},
+		ImageObjects:    []ImageObject{},
+		DocumentObjects: []DocumentObject{},
+	}
+	for {
+		attrs, err := it.Next()
```

Comment:
Perhaps you could try out to run this concurrently with Goroutines.

Reply:
After checking these the [google wiki](https://github.com/googleapis/google-cloud-go/wiki/Iterator-Guidelines) & [a discussion thread](https://stackoverflow.com/questions/63215767/google-cloud-storage-listing-objects-concurrently-go#comment111785255_63215767), it seems a better way is to separate the query when there are too many objects to query.